lib/remote-connector.js

```diff
@@ -70,16 +95,29 @@ RemoteConnector.prototype.define = function(definition) {
 };
 
 RemoteConnector.prototype.resolve = function(Model) {
-  var remotes = this.remotes;
+  var auth = this.auth,
+      remotes = this.remotes,
+      headers = this.headers,
+      headerArgs = this.headerArgs;
 
   Model.sharedClass.methods().forEach(function(remoteMethod) {
     if (remoteMethod.name !== 'Change' && remoteMethod.name !== 'Checkpoint') {
-      createProxyMethod(Model, remotes, remoteMethod);
+      headers.forEach(function(header) {
+        var already = remoteMethod.accepts.filter(function(accept) {
+          return accept.http &&
+              accept.http.source === 'header' &&
+              accept.name === header.name;
+        });
+        if (!already || already.length === 0) {
+          remoteMethod.accepts.push(header);
```

Comment:
This is pretty cool... but I don't think we should be modifying the `remoteMethod` definition. Why not just use the strong-remoting hooks to update the request with headers?

```js
remotes.before('**', function(ctx, next) {
  // modify the headers and call next
  setHeaders(settings.headers, ctx.req);
  next();
});
```